blog/views.py

```diff
@@ -0,0 +1,41 @@
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core.paginator import Paginator
+from django.shortcuts import render, redirect
+from django.views import generic
+
+from blog.models import Post, Commentary
+
+
+def index(request):
+    posts = Post.objects.all().order_by("created_time")
+    paginator = Paginator(posts, 5)
+    page = request.GET.get("page")
+    page_obj = paginator.get_page(page)
+    context = {
+        "posts": page_obj,
+    }
+    return render(request,
+                  "blog/index.html",
+                  context=context, )
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+
+
+@login_required
+def add_comment(request, pk):
+    if request.method == "POST":
+        comment_content = request.POST.get("comment")
```

Comment:
Consider using Django forms for handling form data. It provides validation out of the box and can help prevent common mistakes. For example, it can handle cases where 'comment' key is not present in the POST data.